evm-contracts/src/v0.6/dev/DeviationFlaggingValidator.sol

```diff
@@ -0,0 +1,166 @@
+
+  uint32 constant public THRESHOLD_MULTIPLIER = 100000;
+
+  uint32 public flaggingThreshold;
+  FlagsInterface public flags;
+
+  event FlaggingThresholdUpdated(
+    uint24 indexed previous,
+    uint24 indexed current
+  );
+  event FlagsAddressUpdated(
+    address indexed previous,
+    address indexed current
+  );
+
+  /**
+   * @notice sets up the validator with its threshold and flag address.
+   * @param _flags sets the address of the flags contract
+   * @param _flaggingThreshold sets the threshold that will trigger a flag to be
+   * raised. Setting the value of 100,000 is equivalent to tolerating a 100%
+   * change compared to the previous price.
+   */
+  constructor(
+    address _flags,
+    uint24 _flaggingThreshold
+  )
+    public
+  {
+    setFlagsAddress(_flags);
+    setFlaggingThreshold(_flaggingThreshold);
+  }
+
+  /**
+   * @notice checks whether the parameters count as valid by comparing the
+   * difference change to the flagging threshold.
+   * @param previousRoundId is ignored.
+   * @param previousAnswer is used as the median of the difference with the
+   * current answer to determine if the deviation threshold has been exceeded.
+   * @param roundId is ignored.
+   * @param answer is the latest answer which is compared for a ratio of change
+   * to make sure it has not execeeded the flagging threshold.
+   */
+  function validate(
+    uint256 previousRoundId,
+    int256 previousAnswer,
+    uint256 roundId,
+    int256 answer
+  )
+    external
+    override
+    returns (bool)
+  {
+    if (!isValid(previousRoundId, previousAnswer, roundId, answer)) {
+      flags.raiseFlags(arrayifyMsgSender());
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * @notice checks whether the parameters count as valid by comparing the
+   * difference change to the flagging threshold and raises a flag on the
+   * flagging contract if so. This method conforms to the
+   * AggregatorValidatorInterface.
+   * @param previousAnswer is used as the median of the difference with the
+   * current answer to determine if the deviation threshold has been exceeded.
+   * @param answer is the latest answer which is compared for a ratio of change
+   * to make sure it has not execeeded the flagging threshold.
+   */
+  function isValid(
+    uint256 ,
+    int256 previousAnswer,
+    uint256 ,
+    int256 answer
+  )
+    public
+    view
+    returns (bool)
+  {
+    if (previousAnswer == 0) return true;
+
+    int256 change = previousAnswer.sub(answer);
+    uint256 percent = abs(change.mul(THRESHOLD_MULTIPLIER).div(previousAnswer));
+
+    return percent <= flaggingThreshold;
+  }
+
+  /**
+   * @notice updates the flagging threshold
+   * @param _flaggingThreshold sets the threshold that will trigger a flag to be
+   * raised. Setting the value of 100,000 is equivalent to tolerating a 100%
+   * change compared to the previous price.
+   */
+  function setFlaggingThreshold(uint24 _flaggingThreshold)
```

Comment:
Does that mean we'd make decimals configurable? It's fixed at the moment. Adding more configuration could also be a footgun, as it commonly relies on ignored documentation. An implementation specific concern is also that this could lead to more gas expensive storage reads. I'm leaning towards leaving this as is.

Reply:
I do not think we should make decimals configurable. That would break anyone reading the feed if they aren't checking decimals first. Since decimals hasn't been a part of the AggregatorInterface, no one is currently following such a workflow.